NGitLab/IPackageClient.cs

```diff
@@ -0,0 +1,14 @@
+using NGitLab.Models;
+
+namespace NGitLab
+{
+    public interface IPackageClient
+    {
+        /// <summary>
+        /// Add a package file with the proposed information to the GitLab Generic Package Repository for the selected Project Id.
+        /// </summary>
+        /// <param name="packagePublish">The information about the package file to publish.</param>
+        /// <returns>The package if it was created.  Null if not.</returns>
+        Package Publish(PackagePublish packagePublish);
```

Comment:
New methods must be async. 

```suggestion
        Task<Package> PublishAsync(PackagePublish packagePublish, CancellationToken cancellationToken = default);
```